crates/stc_ts_file_analyzer/src/analyzer/assign/mod.rs

```diff
@@ -2067,7 +2067,21 @@ impl Analyzer<'_, '_> {
                         // TODO: Handle Type::Rest
 
                         if elems.len() > rhs_elems.len() {
-                            return Err(Error::AssignFailedBecauseTupleLengthDiffers { span });
+                            let rhs_elems_cannot_cast = rhs_elems
+                                .iter()
+                                .enumerate()
+                                .filter(|&(i, right_element)| match *(&elems[i]).ty {
+                                    Type::Keyword(KeywordType {
+                                        kind: TsKeywordTypeKind::TsAnyKeyword,
+                                        ..
+                                    }) => true,
+                                    _ => false,
+                                })
+                                .collect::<Vec<_>>();
```

Comment:
this assumption is still not true as:

1. `[1]` is not assignable to`[any, string, string, string]` and should throw an error
2. `[1]` is assignable `[any, any]` and no error should be thrown

So let me do the opposite (with `elems`) to prove it

Reply:
I got rid of it